docs/DeveloperGuide.md

```diff
@@ -1,29 +1,156 @@
 # Developer Guide
+The aim of this guide is to help readers understand how the system and components of RecipEditor 
+is designed, implemented and tested. This developer guide also serves to help 
+developers to understand the architecture of RecipEditor and some design considerations. 
+Click to view the latest release of [RecipEditor]((https://github.com/AY2223S1-CS2113-T18-2/tp/releases)).
+
+## Content page
+[Acknowledgements](#acknowledgements)
+
+[Design](#design)
+- [Architecture](#architecture)
+- [Logic Component](#logic-component)
+- [Ui Component](#ui-component)
+- [Storage Component](#storage-component)
+- [Implementation](#implementation)
+    - [Loading Of Data On Startup](#loading-of-data-on-startup)
+    - [Parsing of Commands](#parsing-of-commands)
+    - [Add Recipe](#add-recipe)
+    - [Edit an Existing Recipe](#add-an-existing-recipe)
+    - [Find Recipe Based on Recipe Name and Ingredient](#find-recipe-based-on-recipe-name-and-ingredient)
+- [Product Scope](#product-scope)
+    - [Target User Profile](#target-user-profile)
+    - [Value Proposition](#value-proposition)
+- [User Stories](#user-stories)
 
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
+### Architecture
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+<p align="center" width="100%">
+  <img width="80%" src="images/MainClassDiagram.png" alt="Storage Class Diagram"/>
+</p>
 
+`Recipeditor` calls to various class a perform all the tasks assigned by the user.
+
+- `Ui`: handles interactions with users, including printing and reading
+- `Storage`: manages the storage of the list of recipes
+- `Command`: instructs the current task to perform
+- `CommandResult`: explains the outcome of each command performed
+- `Parser`: interprets the user input into different commands
+
+Software running flow:
+
+Upon start, Recipeditor will check load or create saves.
```

Comment:
I'm not sure what "check load or create saves" means -- I assume this means it will load saved files if present and create them if they aren't, but this could be made clearer